File: sphere-sdk-java/app/sphere/util/Url.scala
package sphere.util

import scala.collection.JavaConverters._
import java.net.{URL, URLDecoder, URLEncoder}

object Url {
  def buildQueryString(params: Map[String, Any]): String =
    params.filter(_._2 != null).map(p => p._1 + "=" + URLEncoder.encode(p._2.toString, "UTF-8")).mkString("&")

  def parseQueryString(s: String): Map[String, List[String]] = {
    s.split("&").map(_.split("=")).collect {
      case Array(k, v) => (k, URLDecoder.decode(v, "UTF-8"))
    }.groupBy(_._1).map {
      case (k, vs) => k -> vs.map(_._2).toList
    }
  }

  def buildQueryString(params: java.util.Map[String, Any]): String =
    buildQueryString(params.asScala.toMap)
  
  def combine(baseUrl: String, relativeUrl: String): String = {
    return new URL(new URL(baseUrl), relativeUrl).toString();
  }
}

File: sphere-java-client/src/main/java/util/Url.java
package de.commercetools.sphere.client.util;

import java.net.URL;
import java.net.MalformedURLException;

public class Url {
    public static String combine(String baseUrl, String relativeUrl) {
        try {
          return new URL(new URL(baseUrl), relativeUrl).toString();
        } catch (MalformedURLException e) {
            throw new RuntimeException(e);
        }
    }
}
